### DISCUSSION TITLE
Loco Positioning System and Python API

### DISCUSSION THREAD

**Discussion by danielcmdeveloper** [bitcraze/discussions#1547] [General]
Good morning, everyone,
I'm trying to do an experiment with Loco Node Positioning.

The idea is, once the 8 positioning beacons are configured, the idea is to turn them on, turn on the Crazyflie and then initialize the positions of each one of them and be able to do a controlled flight with positioning inside the flight rectangle provided by the positioners.

I want to do all this from the Python API but I can't find a clear example, I only see an example that sets the beacons in bootloader mode.
The configuration specifications of the 8 beacons I have acquired from: https://www.bitcraze.io/documentation/tutorials/getting-started-with-loco-positioning-system/#update-the-nodes

Thank you very much for everything in advance.
Best regards :-)

**Comment by ToveRumar** [comment]
Hi!

This sounds like a fun experiment!

It sounds like what you are struggling with is how to turn on the beacons from the script. Is this correct? 

Unfortunately this is not possible, and you need in that case to connect them to a power source which you can control via the script. Maybe a scriptable USB-hub or remote controlled power plug or something similar could work? 

Once the nodes are configured once, they will keep their configuration until its removed. You have already found a suitable way to configure them right?

If the Crazyflie has a loco deck on board it will detect the beacons when it starts up and will then fly using this positioning. So you can use any positioning example as a base. 

Hope this answer helps you to solve your issue!

**Comment by danielcmdeveloper** [comment]
Good morning,

Thank you very much for your answer, it has been a great help, I think I can start the experiment.
I would like to comment if this is possible (I show an example with only 4 beacons because it could be extrapolated to 6 or 8).

From the Python API I am trying to do the following:

Define the position of the anchors
anchors = [
    {‘id’: 0, ‘x’: 0.0, ‘y’: 0.0, ‘z’: 0.0}, # Anchor 0
    {‘id’: 1, ‘x’: 3.0, ‘y’: 0.0, ‘z’: 0.0}, # Anchor 1
    {‘id’: 2, ‘x’: 0.0, ‘y’: 3.0, ‘z’: 0.0}, # Anchor 2
    {‘id’: 3, ‘x’: 3.0, ‘y’: 3.0, ‘z’: 2.5}, # Anchor 3 (a higher one)
]


def set_anchors(cf):
    for anchor in anchors:
        cf.param.set_value(f'lps.anchor.{anchor[‘id’]}.x‘, anchor[’x'])
        cf.param.set_value(f'lps.anchor.{anchor[‘id’]}.y‘, anchor[’y'])
        cf.param.set_value(f'lps.anchor.{anchor[‘id’]}.z‘, anchor[’z'])
    print(‘Beacon configuration completed.’)

But the following error is displayed: raise KeyError(‘{} not in param TOC’.format(complete_name)) 
KeyError: ‘lps.anchor.0.x not in param TOC’.

Maybe you are not setting the correct variables to set the initial positions?

Thanks again.
Best regards.

**Reply by gemenerik** [reply]
It's saying it cannot find the param you are trying to set in the CF. You can check if it exists in the cfclient parameter tab.
